pkg/instrument/sqlite.go

```diff
@@ -0,0 +1,68 @@
+package instrument
+
+import (
+	"time"
+	_ "unsafe"
+
+	"github.com/mattn/go-sqlite3"
+)
+
+// #cgo LDFLAGS: -lsqlite3
+// #include "extension.h"
+import "C"
+
+// StartSQLiteMonitoring instruments sqlite to collect metrics
+// on all connection opened after the call (i.e. existing connections
+// will not be instrumented).
+// Monitoring must be stopped by calling [StopSQLiteMonitoring].
+func StartSQLiteMonitoring() error {
+	if err := C.sqlite3_instrument(); err != sqlite3.SQLITE_OK {
+		return sqlite3.ErrNo(err)
+	}
+	return nil
+}
+
+// StopSQLiteMonitoring removes instrumentation from sqlite.
+// Existing connections will still be insturmented until they
+// are closed.
```

Comment:
I think we can be more flattering to ourselves here ;) Perhaps the logic could be reversed here?
```
// After this call, no new connections are instrumented.
```